### DISCUSSION TITLE
Will there be a collapsible component?

### DISCUSSION THREAD

**Discussion by MAST1999** [adobe/react-spectrum#6635] [Q&A] [answered]
I was looking for a Collapsible component similar to this component from zag-js https://zagjs.com/components/react/collapsible

BTW, it's kinda off-topic, but I accidentally found that we have an accordion in the repo, but it's not exposed.
https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria/accordion

Is it still a work in progress?

**Comment by yihuiliao** [comment, marked as answer]
Accordion is still in a very early state and we haven't been able to prioritize working on it. You're welcome to try it out but understand that you do so at your own risk since it is still unstable. See https://react-spectrum.adobe.com/Support.html#pre-releases for more information about our versioning.

If you've already taken a look at our docs and haven't found anything like the collapsible component above, you can also check out our [roadmap]( https://github.com/orgs/adobe/projects/19/views/18)

**Reply by MAST1999** [reply]
I couldn't find collapsible on the roadmap.

But this is good enough for now.

Thank you for taking the time to answer ☺️

**Reply by snowystinger** [reply]
A little more reference for why accordion hasn't been prioritized https://github.com/adobe/react-spectrum/issues/1989#issuecomment-1371381252

Also, that collapsible looks like a styled dl/dd/dt, shouldn't need anything from us to make that hopefully.

**Reply by MAST1999** [reply]
Sorry I'm replying a bit late.

The collapsible is a combination of buttons and a div. On button press we hide the div's content and vise versa.

This requires us to implement the disclosure pattern: https://www.w3.org/WAI/ARIA/apg/patterns/disclosure

So it would be nice if it was provided.

Right now I'm using the zag-js's implantation (tweaked the API to make it look like react-aria.)

**Reply by devongovett** [reply]
You can just use the builtin browser one: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details That even has some nice benefits like auto expanding when you use Ctrl+F to search the page.

**Reply by MAST1999** [reply]
I dunno how I forgot about this...

Thanks 👍

**Reply by snowystinger** [reply]
just an fyi, https://react-spectrum.adobe.com/react-aria/Disclosure.html#disclosure is now released
